middlewares/authn.js

```diff
@@ -104,7 +104,7 @@ module.exports = function(req, res, next) {
         return;
     }
     var authnData = csrf.getCheckedData(req.cookies.pl_authn, config.secretKey, {maxAge: 24 * 60 * 60 * 1000});
-    if (authnData == null || authnData.authn_provider_name == null) { // force re-authn if authn_provider_name is missing (for upgrade)
+    if (authnData == null && !/^(\/?)$|^(\/pl\/?)$/.test(req.path)) {
         // if authn cookie check failed then clear the cookie and redirect to login
         res.clearCookie('pl_authn');
         res.redirect('/pl/login');
```

Comment:
```
var authnData = null;
if (req.cookies.pl_authn) {
    // if we have a authn cookie then we try and unpack it
    authnData = csrf.getCheckedData(req.cookies.pl_authn, config.secretKey, {maxAge: 24 * 60 * 60 * 1000});
    // if the cookie unpacking failed then authnData will be null
}
if (authnData == null) {
    // we failed to authenticate
    if (/^(\/?)$|^(\/pl\/?)$/.test(req.path)) {
        // we requested path is the homepage, so allow this request to proceed without an authenticated user
        next();
        return;
    } else {
        // we aren't authenticated, and we've requested some page that isn't the homepage, so bounce to the login page
        // first set the preAuthUrl cookie for redirection after authn
        res.cookie('preAuthUrl', req.originalUrl);
        // clear the pl_authn cookie in case it was bad
        res.clearCookie('pl_authn');
        res.redirect('/pl/login');
        return;
    }
}
```

Reply:
^^^ This code is untested, but I think it's the correct logic. More eyes on this would be good.

Reply:
(I did test a hacked up version which I think was roughly equivalent to this, and it worked)